### DISCUSSION TITLE
SenseCAP Decoder | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1281] [Q&A]
# SenseCAP Decoder | Seeed Studio Wiki

SenseCAP_Decoder

https://wiki.seeedstudio.com/SenseCAP_Decoder/

**Comment by eerke2021** [comment]
Are all the Chirpstack decoder codecs on your GitHub website for Chirpstack V3? I am getting an error using the A1101 one on an SenseCap M2 multi-platform which has chirpstack V4.

**Comment by PeakeElectronicInnovation** [comment]
We are also using Chirpstack V4, managed to get this working by using the Chirpstack V3 decoder, delete the first line:

function Decode(fPort, bytes, variables) {

Replace it with the Chirpstack V4 function name and break out the nested variables so they are available to the decoder functions:

function decodeUplink(input) {
  	var bytes = input.bytes;
  	var fPort = input.fPort;

**Reply by eerke2021** [reply]
As chirpstack V4 and TTN both use a standardised decoder you can also use the TTN decoder. It just looks weird. I found somewhere on their wiki which says to do that.

**Reply by PeakeElectronicInnovation** [reply]
Oh good to know, thanks!

**Comment by cuchty** [comment]
Using S2100 with electricity meter (1000 pulses/kWh) and scaling factor A=0.001, B=0.
The transmitted value is limited to signed int32 (±2,147,483,647), but what about the internal pulse counter "x" before the scaling formula Y=Ax+B is applied?

If internal counter is also signed int32: max ~2.1M pulses = only 2,147 kWh capacity
If internal counter is uint32 or larger: max ~2.1M kWh capacity with A=0.001

Which is it? The documentation doesn't specify the internal counter data type. This is critical for determining the correct scaling factor for long-term deployment.
Can someone from Seeed confirm the internal counter implementation?

**Reply by nfs0619** [reply]
Hello, the maximum measurement value for RS485 on the S2100 is **2,147,483**, which corresponds to the **x** value in the formula *Y = A·x + B*. If this range is too small for your application, you will need to perform data scaling on your side.
If you encounter any other issues during use, please feel free to contact our technical support team at **[sensecap@seeed.cc](mailto:sensecap@seeed.cc)**.

**Comment by freddy36** [comment]
T1000-E TTN/Chirpstack v4 decoder from this page doesn't work.
Code diretly from the github file is working fine:
https://github.com/Seeed-Solution/SenseCAP-Decoder/blob/main/T1000/TTN/SenseCAP_T1000E_TTN_Decoder.js

**Reply by 1cecO1** [reply]
Hello, the T1000 decoder displayed on this wiki page is for the T1000-A/B versions, not the T1000-E version.

Please find the T1000-E for LoRaWAN from this page:
https://wiki.seeedstudio.com/get_started_with_lorawan_tracker/#decoder
